source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -544,14 +547,24 @@ class WordDocument(UIADocumentWithTableNavigation,WordDocumentNode,WordDocumentB
 	def event_textChange(self):
 		# Ensure Braille is updated when text changes,
 		# As Microsoft Word does not fire caret events when typing text, even though the caret does move.
+		# Update braille also when tethered to review, and review position
+		# if review follows caret.
 		braille.handler.handleCaretMove(self)
+		review.handleCaretMove(self)
 
 	def event_UIA_notification(self, activityId=None, **kwargs):
 		# #10851: in recent Word 365 releases, UIA notification will cause NVDA to announce edit functions
 		# such as "delete back word" when Control+Backspace is pressed.
 		if activityId == "AccSN2":  # Delete activity ID
 			return
 		super(WordDocument, self).event_UIA_notification(**kwargs)
+		# Try to ensure that braille is updated when UIA is not used and

```

Comment:
I don't think it is a good idea to do this here. There must be another event we're missing, and if not, it's simply a bug in Word that needs to be reported and fixed on their end.

Reply:
Do you intend this to work for Non UIA cases with this code? If so I think it's a false positive, because this code is not active when UIA is disabled.

Reply:
When "Use UI Automation to access Microsoft Word document controls Only when necessary" event_UIA_notification is fired when pressing control+x and control+z.

Can reason be that object model is not available for some reason and thus UIA is used?

Reply:
If a notification event is fired for undo but neither a caret nor a text change event is fired, this is something Microsoft needs to fix.